tests/UI/campaigns/regression/currencies/computingPrecision_FO.js

```diff
@@ -0,0 +1,535 @@
+require('module-alias/register');
+
+// Helpers to open and close browser
+const helper = require('@utils/helpers');
+
+// Common tests login BO
+const loginCommon = require('@commonTests/loginBO');
+
+// Import pages
+
+// BO pages
+const dashboardPage = require('@pages/BO/dashboard');
+const cartRulesPage = require('@pages/BO/catalog/discounts');
+const addCartRulePage = require('@pages/BO/catalog/discounts/add');
+const localizationPage = require('@pages/BO/international/localization');
+const currenciesPage = require('@pages/BO/international/currencies');
+const addCurrencyPage = require('@pages/BO/international/currencies/add');
+const ordersPage = require('@pages/BO/orders');
+const sqlManagerPage = require('@pages/BO/advancedParameters/database/sqlManager');
+const addSqlQueryPage = require('@pages/BO/advancedParameters/database/sqlManager/add');
+const viewSqlQueryPage = require('@pages/BO/advancedParameters/database/sqlManager/view');
+
+// FO pages
+const homePage = require('@pages/FO/home');
+const searchResultsPage = require('@pages/FO/searchResults');
+const foLoginPage = require('@pages/FO/login');
+const productPage = require('@pages/FO/product');
+const cartPage = require('@pages/FO/cart');
+const checkoutPage = require('@pages/FO/checkout');
+const orderConfirmationPage = require('@pages/FO/checkout/orderConfirmation');
+
+
+// Import expect from chai
+const {expect} = require('chai');
+
+// Import demo data
+const {PaymentMethods} = require('@data/demo/paymentMethods');
+const {DefaultAccount} = require('@data/demo/customer');
+const {Currencies} = require('@data/demo/currencies');
+
+const {Products} = require('@data/demo/products');
+
+
+// Import faker data
+const CartRuleFaker = require('@data/faker/cartRule');
+
+const percentCartRule = new CartRuleFaker(
+  {
+    name: 'discount15',
+    code: 'discount15',
+    discountType: 'Percent',
+    discountPercent: 15,
+  },
+);
+
+const giftCartRule = new CartRuleFaker(
+  {
+    name: 'freeGiftMug',
+    code: 'freeMug',
+    discountType: 'None',
+    freeGift: true,
+    freeGiftProduct: Products.demo_13,
+  },
+);
+
+// Create sql query data to get last order discount and total price
+const sqlQueryData = {
+  name: 'Discount and TTC from last order',
+  sqlQuery: 'SELECT total_discounts, total_paid_tax_incl '
+    + 'from  ps_orders '
+    + 'ORDER BY id_order '
+    + 'DESC LIMIT 1;',
```

Comment:
Is there a way to make this query more resilient ? I'm thinking about launching differents campaigns in parallel and maybe having an order inserted at the wrong moment... Maybe adding the order reference that you get after creating it ?

Reply:
Now we get order reference from BO order page to use it in sql query